src/main/java/org/pf4j/update/PluginInfo.java

```diff
@@ -45,19 +46,19 @@
     private String repositoryId;
 
     // Cache lastRelease per system version
-    private Map<Version, PluginRelease> lastRelease = new HashMap<>();
+    private Map<String, PluginRelease> lastRelease = new HashMap<>();
 
     /**
      * Returns the last release version of this plugin for given system version, regardless of release date
      * @param systemVersion version of host system where plugin will be installed
      * @return PluginRelease which has the highest version number
      */
-    public PluginRelease getLastRelease(Version systemVersion) {
+    public PluginRelease getLastRelease(String systemVersion, VersionManager versionManager) {
```

Comment:
I don't know if useful to add `VersionManager` as parameter. The actual `satisfies`method from VersionManager works with String and here we have `Expression` from semver library.